### DISCUSSION TITLE
Creation d'un nouveau service : Antvoice

### DISCUSSION THREAD

**Discussion by CaroS77** [AmauriC/tarteaucitron.js#962] [Ideas]
Bonjour,

Nous allons passer sur la version payante de TAC.

J'ai vu que vous pouviez ajouter des services non existants. Vous serait-il possible d'ajouter le service suivant : Antvoice 
(https://www.antvoice.com/fr/ - Partner display)

**voici les éléments pour la création**
- le marqueur html => <script async src="https://static.avads.net/avtag.min.js"></script>

- la liste des cookies déposés
av-mid => www.ads.avads.net
av-sess-id-335 => www.ads.avads.net
av-tp-bsw => www.ads.avads.net

Finalité :
mid : c'est l'identifiant du navigateur avec une durée d'un mois
sess-id : c'est un id de session qui a une durée courte de l'ordre d'1h
tp : ce sont des cookies de synchro avec nos partenaires (google ou bidswitch) qui durent environ une semaine

 - le lien de la politique de confidentialité : https://www.antvoice.com/fr/privacy-policy/

Je vous remercie d'avance et vous félicite pour ce module.

CaroS77

**Comment by AmauriC** [comment]
Merci, je viens de l'ajouter : )
https://github.com/AmauriC/tarteaucitron.js/commit/6e7528dcd04a64a43193be61eabdd7e74496eb5b

**Reply by CaroS77** [reply]
Bonjour Amauri,

Le marqueur html / initialisation du script à utiliser pour Antvoice est en fait le suivant : 

<script>
window.avDataLayer = window.avDataLayer || [];
window.avtag = window.avtag ||
function(_cmd,_p){window.avDataLayer.push({cmd:_cmd,p:_p});}
avtag('init', {id:'XXXX'});
</script>
<script async src="https://static.avads.net/avtag.min.js"></script>

Je vous remercie d'avance de votre action pour la modification de celui-ci
Bien cdlt

**Comment by CaroS77** [comment]
Super, merci Amaury

**Comment by CaroS77** [comment]
Bonjour Amauri,

Excusez-moi, ce script n'était toujours pas complet.
Il faut ajouter le script de consentement ci-dessous

<script>
window.avDataLayer = window.avDataLayer|| [];
window.avtag = window.avtag ||
function(_cmd,_p){window.avDataLayer.push({cmd:_cmd,p:_p});}
avtag('setConsent', {consent:true});
</script>

Ce doit donc être :

//Initialisation du script
<script>
window.avDataLayer = window.avDataLayer || [];
window.avtag = window.avtag ||
function(_cmd,_p){window.avDataLayer.push({cmd:_cmd,p:_p});}
avtag('init', {id:'XXX'});
</script>
<script async src="https://static.avads.net/avtag.min.js"></script>

//Collecte du consentement de l’internaute
<script>
window.avDataLayer = window.avDataLayer|| [];
window.avtag = window.avtag ||
function(_cmd,_p){window.avDataLayer.push({cmd:_cmd,p:_p});}
avtag('setConsent', {consent:true});
</script>

Je vous remercie d'avance,
Caro

**Reply by AmauriC** [reply]
Ca a été ajouté : https://github.com/AmauriC/tarteaucitron.js/blob/master/tarteaucitron.services.js#L110
